### PR TITLE
news: add notes for 2.9.0

### PR DESCRIPTION
**Pull request by sohankunkerkar**

### CODE REVIEW

File: NEWS [hunk 1]
```diff
@@ -1,3 +1,18 @@
+08-Jan-2021 Ignition 2.9.0
+
+  Changes:
+
+    - Require filesystems.format if wipeFilesystem or mountOptions is specified
+    - Refactor code to address golangci-lint warnings
+
+  Bug Fixes:
+
+    - Fix fetching configs from S3 resources when running on non-default AWS partitions
```
Comment (bgilbert) [inline NEWS:10]: Line-wrap to 80 chars.

File: NEWS [hunk 2]
```diff
@@ -1,3 +1,18 @@
...
+    - Support fetching userdata from IMDSv2 on AWS
+    - Fix crash on partitions with no number or label
+    - Correctly document storage.filesystems.path as optional
+    - Clarify semantics of systemd.units.name
```
Comment (bgilbert) [inline NEWS:14]: Since this is just a docs change, `Clarify documented semantics` would be clearer.

File: NEWS [hunk 3]
```diff
@@ -1,3 +1,18 @@
+08-Jan-2021 Ignition 2.9.0
+
+  Changes:
+
+    - Require filesystems.format if wipeFilesystem or mountOptions is specified
```
Comment (bgilbert) [inline NEWS:5]: I guess we usually write the full path, `storage.filesystems.format`.

File: NEWS [hunk 4]
```diff
@@ -1,3 +1,18 @@
...
+    - Refactor code to address golangci-lint warnings
+
+  Bug Fixes:
+
+    - Fix fetching configs from S3 resources when running on non-default AWS partitions
+    - Support fetching userdata from IMDSv2 on AWS
```
Comment (bgilbert) [inline NEWS:11]: Hmm.  If this was new functionality, it should go under Features.  But 2.4.0 claimed to support IMDSv2, so actually your original "Fix" phrasing was the right way to go.  "Fix userdata fetch from IMDSv2 on AWS"?

Comment (sohankunkerkar) [reply NEWS]: ok

**Review by bgilbert — APPROVED** [review]

**Pull request merged**